--- before.php	2010-06-15 08:37:38.000000000 +0100
+++ after.php	2010-06-15 08:34:02.000000000 +0100
@@ -11,7 +11,7 @@
 ?>
 				<table id="caption-date-table">
 					<tr>
-						<td><h2 id="picture-caption"><?php echo plogger_get_picture_caption(); ?></h2></td>
+						<td><h2 id="picture-caption" style="cursor:pointer" onclick="document.location.href='/'"><?php echo plogger_get_picture_caption(); ?></h2></td>
 						<td class="align-right"><h2 class="date"><?php echo plogger_get_picture_date()?></h2></td>
 					</tr>
 				</table><!-- /caption-date-table -->
@@ -24,12 +24,16 @@
 				</table><!-- /prev-next-table -->
 
 				<div id="picture-holder">
-					<a accesskey="v" href="<?php echo plogger_get_source_picture_url(); ?>"><img class="photos-large" src="<?php echo plogger_get_picture_thumb(THUMB_LARGE); ?>" width="<?php echo $thumb_width; ?>" height="<?php echo $thumb_height; ?>" title="<?php echo plogger_get_picture_caption('clean'); ?>" alt="<?php echo plogger_get_picture_caption('clean'); ?>" /></a>
-					<p id="description"><?php echo plogger_get_picture_description(); ?></p>
-					<p id="exif-toggle"><?php echo plogger_get_detail_link(); ?></p>
-					<div id="exif-data-container">
-<?php echo generate_exif_table(plogger_get_picture_id()); ?>
-					</div><!-- /exif-data-container -->
+					<?php $desc = plogger_get_picture_description(); ?>
+					<?php if (strlen($desc) > 8 && 0 == substr_compare($desc,"YouTube:",0,8)) : ?> 
+	<?php $ytcode = substr(plogger_get_picture_description(), 8); ?>
+	<a href="http://youtube.com/watch?v=<?php echo $ytcode; ?>" target="_blank">
+	<object width="640" height="385"><param name="movie" value="http://www.youtube.com/v/<?php echo $ytcode; ?>&hl=en_GB&fs=1&"></param><param name="allowFullScreen" value="true"></param><param name="allowscriptaccess" value="always"></param><embed src="http://www.youtube.com/v/<?php echo $ytcode; ?>&hl=en_GB&fs=1&" type="application/x-shockwave-flash" allowscriptaccess="always" allowfullscreen="true" width="640" height="385"></embed></object>
+	</a>
+<?php else : ?>
+	<a accesskey="v" href="<?php echo plogger_get_source_picture_url(); ?>"><img class="photos-large" src="<?php echo plogger_get_picture_thumb(THUMB_LARGE); ?>" width="<?php echo $thumb_width; ?>" height="<?php echo $thumb_height; ?>" title="<?php echo plogger_get_picture_caption('clean'); ?>" alt="<?php echo plogger_get_picture_caption('clean'); ?>" /></a>
+	<p id="description"><?php echo plogger_get_picture_description(); ?></p>
+<?php endif; ?>
 				</div><!-- /picture-holder -->
 
 			</div><!-- /big-picture-container -->
@@ -44,4 +48,4 @@
 <?php endif; ?>
 		</div><!-- /inner-wrapper -->
 
-<?php plogger_get_footer(); ?>
\ No newline at end of file
+<?php plogger_get_footer(); ?>
